### PR TITLE
chore: add F1 score example of pytorch custom reducers [DET-4724]

### PR DESCRIPTION
**Pull request by rb-determined-ai**
## Description

This change also includes renaming the `data_layer` category of examples into a `features` category, which now includes a the custom reducers example.

## Test Plan

A new test was added for the new example.

### CODE REVIEW

File: examples/features/custom_reducers_mnist_pytorch/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# PyTorch Custom Reducers (MNIST)
+This tutorial shows how to use custom reducers with PyTorch.  In this example,
+the custom reducer is a per-class F1 score.
+
+This example is based on Determined's `mnist_pytorch` tutorial, with the custom
+reducer as the only modification.
+
+## Files
+* **model_def.py**: The .
```
Comment (naren-determined) [inline examples/features/custom_reducers_mnist_pytorch/README.md:9]: Fill this in :)

File: examples/features/custom_reducers_mnist_pytorch/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+## To Run
+If you have not yet installed Determined, installation instructions can be found
+under `docs/install-admin.html` or at https://docs.determined.ai/latest/index.html
+
+Run the following command: `det -m <master host:port> experiment create -f
+const.yaml .`. The other configurations can be run by specifying the appropriate
+configuration file in place of `const.yaml`.
+
+## Results
+You should see the per-class F1 scores the Determined WebUI as well as while
```
Comment (naren-determined) [inline examples/features/custom_reducers_mnist_pytorch/README.md:21]: Grammar nit: `You should see the per-class F1 scores **in** the Determined Webui`. Can you just note here results will be the same as the mnist_pytorch example?

File: examples/features/custom_reducers_mnist_pytorch/const.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+description: mnist_pytorch_const
```
Comment (naren-determined) [inline examples/features/custom_reducers_mnist_pytorch/const.yaml:1]: Can you change the name to have custom reducers here in it?

File: examples/features/custom_reducers_mnist_pytorch/distributed.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+description: mnist_pytorch_distributed
```
Comment (naren-determined) [inline examples/features/custom_reducers_mnist_pytorch/distributed.yaml:1]: Ditto on the name.

File: examples/features/custom_reducers_mnist_pytorch/model_def.py [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+The parts of the file which differ from the tutorial are the PerClassF1Score class
+and the calls to wrap_reducer() and self.f1_score.update.
+"""
+
+from typing import Any, Dict, Sequence, Tuple, Union, cast
+
+import torch
+from torch import nn
+
+from layers import Flatten  # noqa: I100
```
Comment (naren-determined) [inline examples/features/custom_reducers_mnist_pytorch/model_def.py:15]: What's this comment here?

**Review by naren-determined — COMMENTED** [review]
I think you may also need to update `docs/examples.txt` as well, but otherwise address these and it looks good!

File: examples/features/custom_reducers_mnist_pytorch/model_def.py (hunk 5 above)
Comment (rb-determined-ai) [reply examples/features/custom_reducers_mnist_pytorch/model_def.py:15]: It's for the linter, but it's copy/pasted from the original

File: examples/features/custom_reducers_mnist_pytorch/const.yaml (hunk 3 above)
Comment (rb-determined-ai) [reply examples/features/custom_reducers_mnist_pytorch/const.yaml:1]: done

File: examples/features/custom_reducers_mnist_pytorch/distributed.yaml (hunk 4 above)
Comment (rb-determined-ai) [reply examples/features/custom_reducers_mnist_pytorch/distributed.yaml:1]: done

File: examples/features/custom_reducers_mnist_pytorch/README.md (hunk 1 above)
Comment (rb-determined-ai) [reply examples/features/custom_reducers_mnist_pytorch/README.md:9]: done:

```
**model_def.py**: Where the custom reducer is defined and used.
```

File: examples/features/custom_reducers_mnist_pytorch/README.md (hunk 2 above)
Comment (rb-determined-ai) [reply examples/features/custom_reducers_mnist_pytorch/README.md:21]: done

**Review by naren-determined — APPROVED** [review]
Looks great! 🚢

**Pull request merged**